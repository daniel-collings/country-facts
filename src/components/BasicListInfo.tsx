import React from 'react'

interface IBasicListInfoProps {
  label: string
  items: string[]
}

export default function BasicListInfo({ label, items }: IBasicListInfoProps) {
  return (
    <div className="p-4 grid">
      <span className="font-bold">{label}</span>
      <ul>
        {items.map((item, index) => (
          <li key={index}>{item}</li>
        ))}
      </ul>
    </div>
  )
}
