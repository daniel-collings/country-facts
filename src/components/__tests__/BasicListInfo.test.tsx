import { describe, expect, test } from 'vitest'
import { render } from '@/utils/testUtils.tsx'
import BasicListInfo from '@/components/BasicListInfo.tsx'

describe('BasicListInfo Component', () => {
  test('should render correctly', () => {
    const { asFragment } = render(
      <BasicListInfo items={['ESP', 'AND', 'DEU']} label="Borders" />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
