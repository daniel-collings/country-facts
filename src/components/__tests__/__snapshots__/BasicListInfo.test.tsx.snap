// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BasicListInfo Component > should render correctly 1`] = `
<DocumentFragment>
  <div
    class="h-full min-h-screen bg-base-100 p-4 sm:p-8"
    data-theme="light"
  >
    <main
      class="max-w-7xl mx-auto px-0 sm:px-6 lg:px-8"
    />
  </div>
</DocumentFragment>
`;
